tools/riscv-openocd-ci/download_toolchain.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+TOOLCHAIN_URL="https://buildbot.embecosm.com/job/riscv32-gcc-ubuntu1804"
```

Comment:
This is definitely going to cause trouble. The test suite uses riscv64 because it's possible to build both 32- and 64-bit binaries with that, but the same is (was?) not true of the 32-bit tools.

Reply:
Actually, it should be all right. The Embecosm's toolchain build can compile for both `rv32` and `rv64`, depending on `-march`. It is my understanding that the prefix (`riscv32-...`) mostly indicates what the [default march](https://github.com/embecosm/embecosm-toolchain-releases/blob/master/stages/build-riscv32-gcc.sh#L14) is should anyone omit that argument, otherwise it is of no importance.

Reply:
> It is my understanding that the prefix (riscv32-...) mostly indicates what the default march is should anyone omit that argument, otherwise it is of no importance.

Just to be pedantic I think you mean default XLEN and not march since the march is the XLEN plus any expected extensions (rv<XLEN><extensions> e.g. rv64gc). I was also under the impression that most riscv32 toolchains could only generate 32 bit code while most riscv64 toolchains could generated 32 or 64 bit code depending on the XLEN specified in the march but I could be out of touch with the details and, as you say, it looks like there's at least one riscv32 toolchain that can generated both 32 and 64 bit code...

Reply:
@TommyMurphyTM1234 - As for the terminology, you are right, XLEN is more precise here.

As for the compiler capabilities, you can check them by a simple experiment, for instance:

```
$ cat test.c
#include <stdint.h>

int main()
{
	*((uint64_t*)0xaabb0000) = 0x1122334455667788;
	return 0;
}
$ ./riscv32-unknown-elf-gcc -Wall -march=rv64gc -mabi=lp64d -c test.c -o /tmp/test.o
$ ./riscv32-unknown-elf-objdump -d -S /tmp/test.o
```